Makefile

```diff
@@ -20,14 +21,27 @@ lint:
 	@type golangci-lint > /dev/null || ( echo "golangci-lint not found. Install it manually or by running 'make setup_lint'."; exit 1 )
 	golangci-lint run
 
+build: $(BUILD_TARGETS)
+
 install: all
-	@$(foreach d,$(SRCDIRS), cd $(ROOT_DIR)/$(d); cp $(shell basename $(d)) ~/go/bin;)
+	@$(foreach d,$(SRCDIRS), cd $(ROOT_DIR)/$(d); cp $(shell basename $(d)) /go/bin;)
+
+integration: install $(INTEGRATION_TARGETS)
 
-# using eval to create as many rules as we have $TARGETS
+# using eval to create as many rules as we have $BUILD_TARGETS
 # each target corresponds to the binary file name (e.g. sensorapp/sensorserver/sensorserver)
 define gobuild_tmpl =
 .PHONY: $(1)
 $(1): go.mod $(2)
 	go build -o $$(dir $$@) ./$$(dir $$@)
 endef
-$(foreach D,$(TARGETS),$(eval $(call gobuild_tmpl, $(D), $(shell find $(dir $(D)) -name '*.go') )))
+$(foreach D,$(BUILD_TARGETS),$(eval $(call gobuild_tmpl, $(D), $(shell find $(dir $(D)) -name '*.go') )))
+
+# using eval to create as many rules as we have $INTEGRATION_TARGETS
+# each target corresponds to the binary file name, appended with `_integration`(e.g. sensorapp/sensorserver/sensorserver_integration)
+define gotest_tmpl =
+.PHONY: $(1)
+$(1): go.mod $(2)
+	go test -v -tags=integration ./$$(dir $$@)
+endef
+$(foreach D,$(INTEGRATION_TARGETS),$(eval $(call gotest_tmpl, $(D), $(shell find $(dir $(D)) -name '*.go') )))
```

Comment:
Having a separate target per integration test enables easily running a specific integration test, since running all integration tests may take longer than acceptable when running interactively.

Reply:
The setup with `go test` makes it quite trivial to run the same command for a subset, e.g. `go test -tags=integration ./netcat/`. I don't think this needs to be in the makefile. 

_If_ we put it in the Makefile, I would very strongly suggest to explicitly list each individual target instead of this macro madness.